shared/src/commonMain/kotlin/co/nimblehq/kmm/template/data/repository/RepositoryImpl.kt

```diff
@@ -0,0 +1,16 @@
+package co.nimblehq.kmm.template.data.repository
+
+import co.nimblehq.kmm.template.data.remote.datasources.UserRemoteDataSource
+import co.nimblehq.kmm.template.data.remote.model.response.toModel
+import co.nimblehq.kmm.template.domain.model.UserModel
+import co.nimblehq.kmm.template.domain.repositories.Repository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flow
+
+class RepositoryImpl(
+    private val userRemoteDataSource: UserRemoteDataSource
+) : Repository {
+    override fun getUsers(): Flow<List<UserModel>> = flow {
+        userRemoteDataSource.getUsers().map { it.toModel() }
+    }
```

Comment:
Should we apply a [ResponseMapping](https://github.com/luongvo/kmm-survey/blob/develop/shared/src/commonMain/kotlin/vn/luongvo/kmm/survey/data/extensions/ResponseMapping.kt) logic to catch and map errors in the network layer?


```suggestion
    override fun getUsers(): Flow<List<UserModel>> = flowTransform {
        userRemoteDataSource.getUsers().map { it.toModel() }
    }
```

Reply:
Please don't forget a test file for it 😄 https://github.com/luongvo/kmm-survey/blob/develop/shared/src/commonTest/kotlin/vn/luongvo/kmm/survey/data/extensions/ResponseMappingTest.kt